File: src/main/java/com/maxdev/warehouse/models/Rack.java
package com.maxdev.warehouse.models;

import jakarta.persistence.*;

@Entity
@Table(name="racks")
public class Rack {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String addr;
    @OneToOne
    @JoinColumn(name="good")
    private Good good;
    private Integer number;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getAddr() {
        return addr;
    }

    public void setAddr(String addr) {
        this.addr = addr;
    }

    public Good getGood() {
        return good;
    }

    public void setGood(Good good) {
        this.good = good;
    }

    public Integer getNumber() {
        return number!=null?number:0;
    }

    public void setNumber(Integer number) {
        this.number = number;
    }
}

File: warehouse-app/src/services/openapi/models/Rack.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Good } from './Good';

export type Rack = {
    id?: number;
    addr?: string;
    good?: Good;
    number?: number;
};
export function rackToString(el:Rack){
    // const goodField =(!!el.good && "name" in el.good)?el.good.name:'null';
    return el&&el.addr?el.addr:'...';
}
